### ISSUE TITLE
[JIT] [Script] not support short-cut eval in And

### ISSUE THREAD

**Issue by wanchaol** [pytorch/pytorch#11042]
```python
@torch.jit.script
def testAnd(t):
    i = 5
    if False and t[i]:
        ret = t[i]
    else:
        ret = t[0]
    return ret

testAnd(torch.randn(4))
```

error:

```

RuntimeError:
select(): index 5 out of range for tensor of size [4] at dimension 0:
operation failed in interpreter:
@torch.jit.script
def testAnd(t):
    i = 5
    if False and t[i]:
                 ~~~ <--- HERE
        ret = t[i]
    else:
        ret = t[0]
    return ret
```

**Comment by eellison** [comment]
I am working on this

**Comment by wanchaol** [comment]
fixed by #11116
